python/sparkdl/image/imageIO.py

```diff
@@ -26,90 +25,61 @@
 from pyspark import SparkContext
 from pyspark.sql.types import (BinaryType, IntegerType, StringType, StructField, StructType)
 from pyspark.sql.functions import udf
+from pyspark.sql.column import Column
+from pyspark.sql.column import _to_seq
+from pyspark.sql.column import _to_java_column
 
+from pyspark.ml.image import ImageSchema
 
-imageSchema = StructType([StructField("mode", StringType(), False),
-                          StructField("height", IntegerType(), False),
-                          StructField("width", IntegerType(), False),
-                          StructField("nChannels", IntegerType(), False),
-                          StructField("data", BinaryType(), False)])
 
-
-# ImageType class for holding metadata about images stored in DataFrames.
+# ImageType represents supported OpenCV types
 # fields:
+#   name - OpenCvMode
+#   ord - Ordinal of the corresponding OpenCV mode (stored in mode field of ImageSchema).
 #   nChannels - number of channels in the image
-#   dtype - data type of the image's "data" Column, sorted as a numpy compatible string.
-#   channelContent - info about the contents of each channel currently only "I" (intensity) and
-#     "RGB" are supported for 1 and 3 channel data respectively.
-#   pilMode - The mode that should be used to convert to a PIL image.
-#   sparkMode - Unique identifier string used in spark image representation.
-ImageType = namedtuple("ImageType", ["nChannels",
-                                     "dtype",
-                                     "channelContent",
-                                     "pilMode",
-                                     "sparkMode",
-                                     ])
-class SparkMode(object):
-    RGB = "RGB"
-    FLOAT32 = "float32"
-    RGB_FLOAT32 = "RGB-float32"
-
-supportedImageTypes = [
-    ImageType(3, "uint8", "RGB", "RGB", SparkMode.RGB),
-    ImageType(1, "float32", "I", "F", SparkMode.FLOAT32),
-    ImageType(3, "float32", "RGB", None, SparkMode.RGB_FLOAT32),
-]
-pilModeLookup = {t.pilMode: t for t in supportedImageTypes
-                 if t.pilMode is not None}
-sparkModeLookup = {t.sparkMode: t for t in supportedImageTypes}
-
-
-def imageArrayToStruct(imgArray, sparkMode=None):
+#   dtype - data type of the image's array, sorted as a numpy compatible string.
+OcvType = namedtuple("OcvType",["name","ord","nChannels","dtype"])
+
+supportedOcvTypes = (
```

Comment:
actually it's a super set. Spark does not have the float32 types which we need.

Reply:
Is there a scenario where as an output of a tf graph we end up with images that are one of the float32 types, save the output as images, then load them again with Spark? Would everything be okay because Spark would essentially "upcast" to float64 from float32?

Reply:
Hmm, interesting point. I don't think there is at the moment since there is no image handling on scala side (i.e. no imagestruct <-> imagearray conversions). I will double check though.